plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoSession.java

```diff
@@ -193,7 +205,12 @@ public void dropTable(SchemaTableName tableName)
     }
 
     public void addColumn(SchemaTableName schemaTableName, ColumnMetadata columnMetadata)
+            throws Exception
     {
+        if (columnMetadata.getName().contains(".")) {
+            throw new Exception("Columns with ' . ' are not supported ");
+        }
+
```

Comment:
Please extract a method since we need the validation in several places. Also, `$` should be also disallowed. By the way, the operation we would like to disallow those special characters is `INSERT` rather than `ADD COLUMN`.